blocks/loops.ts

```diff
@@ -342,18 +348,16 @@ const CONTROL_FLOW_IN_LOOP_CHECK_MIXIN = {
   /**
    * Called whenever anything on the workspace changes.
    * Add warning if this flow block is not nested inside a loop.
-   * @param {!AbstractEvent} e Move event.
-   * @this {Block}
    */
-  onchange: function(e) {
+  onchange: function(this: ControlFlowInLoopBlock, e: Abstract) {
+    const ws = this.workspace as WorkspaceSvg;
     // Don't change state if:
     //   * It's at the start of a drag.
     //   * It's not a move event.
-    if (!this.workspace.isDragging || this.workspace.isDragging() ||
-        e.type !== Events.BLOCK_MOVE) {
+    if (!ws.isDragging || ws.isDragging() || e.type !== Events.BLOCK_MOVE) {
       return;
     }
-    const enabled = this.getSurroundLoop(this);
+    const enabled = !!this.getSurroundLoop();
```

Comment:
Good for clarity!  I'm curious: was this necessary to get the ternary below to type-check?